src/library/scala/collection/ArrayOps.scala

```diff
@@ -221,7 +221,7 @@ final class ArrayOps[A](val xs: Array[A]) extends AnyVal {
     *        x >  0       if this.length >  len
     *   }}}
     */
-  def lengthCompare(len: Int): Int = xs.length - len
```

Comment:
what's the scenario where this fails?  The test below has Int.MinValue, which seems impossible for a length...

Reply:
it passes in `Int.MinValue` as the value of `len`, which is perfectly possible. See [this comment](https://github.com/scala/scala/pull/6930#discussion_r202345125).

Reply:
Well OK, but why would length be negative?

Reply:
because someone passed a negative number in ¯\\\_(ツ)_/¯

Reply:
@isaacl - As a concrete example, suppose you have a collection `xs` that you will use to pair with any overflow of `ys` which is normally going to be paired with `zs`.  You might want to know if `xs` is big enough.  So you ask `xs.lengthCompare(ys.length - zs.length)`.

Beyond that, even if we can't think of a sensible use case for a negative value being passed in, it still should return the logically correct result, because we can't always anticipate what someone might try to use it for.

Reply:
OK.  Well, this function doesn't feel super useful to me, but I guess if there's no preconditions it should return correctly.